File: exercises/small_problems/easy5/letter_counter1.rb
def word_sizes(str)
  size_hash = {}
  str_arr = str.split.map! { |word| word.size }

  str_arr.each do |num|
    size_hash.include?(num) ? size_hash[num] += 1 : size_hash[num] = 1
  end

  size_hash
end

p word_sizes('Four score and seven.') == { 3 => 1, 4 => 1, 5 => 1, 6 => 1 }
p word_sizes('Hey diddle diddle, the cat and the fiddle!') == { 3 => 5, 6 => 1, 7 => 2 }
p word_sizes("What's up doc?") == { 6 => 1, 2 => 1, 4 => 1 }
p word_sizes('') == {}

File: exercises/jsSmallProblems/easy3/letterCounter1.js
const wordSizes = (string) => {
  const result = {};
  if (string.trim().length === 0) return {};

  const lengths = string.split(' ').map((word) => word.length);

  for (let i = 0; i < lengths.length; i += 1) {
    result[lengths[i]] = result[lengths[i]] || 0;
    result[lengths[i]] += 1;
  }

  return result;
};


console.log(wordSizes('Four score and seven.')); // { "3": 1, "4": 1, "5": 1, "6": 1 }
console.log(wordSizes('Hey diddle diddle, the cat and the fiddle!')); // { "3": 5, "6": 1, "7": 2 }
console.log(wordSizes("What's up doc?")); // { "2": 1, "4": 1, "6": 1 }
console.log(wordSizes('')); // {}
